pkg/proto/merge/merge_test.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package merge
+
+import (
+	"testing"
+
+	listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"istio.io/istio/pkg/test/util/assert"
+)
+
+func TestMerge(t *testing.T) {
```

Comment:
This test passes without the rest of the PR. So its probably not testing quite the right thing?

BTW this issue is widespread, this is why we do this everywhere: https://github.com/istio/istio/blob/f524f6319dab722faaeee13d91fd34ab8732c664/pilot/pkg/networking/core/cluster_builder.go#L737

Reply:
SGTM, let me remove

Reply:
@howardjohn Chnaged the test too, should check the dst.